src/main/java/net/shadowmage/ancientwarfare/structure/block/BlockBrazierEmber.java

```diff
@@ -0,0 +1,230 @@
+		return false;
+	}
+
+	@Override
+	public int getLightValue(IBlockState state, IBlockAccess world, BlockPos pos) {
+		return state.getValue(LIT) ? 6 : 0;
+	}
+
+	@Override
+	public BlockFaceShape getBlockFaceShape(IBlockAccess worldIn, IBlockState state, BlockPos pos, EnumFacing face) {
+		return BlockFaceShape.UNDEFINED;
+	}
+
+	@Override
+	public boolean canPlaceBlockAt(World worldIn, BlockPos pos) {
+		return worldIn.getBlockState(pos.down()).isSideSolid(worldIn, pos.down(), EnumFacing.UP);
+	}
+
+	@Override
+	public void onBlockPlacedBy(World worldIn, BlockPos pos, IBlockState state, EntityLivingBase placer, ItemStack stack) {
+		if (!stack.hasTagCompound()) {
+			return;
+		}
+		//noinspection ConstantConditions
+		state = state
+				.withProperty(LIT, stack.getTagCompound().getBoolean(LIT_TAG));
+		worldIn.setBlockState(pos, state);
+	}
+
+	@Override
+	public ItemStack getPickBlock(IBlockState state, RayTraceResult target, World world, BlockPos pos, EntityPlayer player) {
+		ItemStack pickStack = new ItemStack(Item.getItemFromBlock(this));
+		pickStack.setTagCompound(new NBTBuilder().setBoolean(LIT_TAG, state.getValue(LIT)).build());
+		return pickStack;
+	}
+
+	@Override
+	@SideOnly(Side.CLIENT)
+	public BlockRenderLayer getBlockLayer() {
+		return BlockRenderLayer.CUTOUT;
+	}
+
+	@Override
+	public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, EntityPlayer player, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ) {
+		ItemStack stack = player.getHeldItem(hand);
+		if (!state.getValue(LIT) && isFireStarter(stack)) {
+			if (world.isRemote) {
+				return true;
+			}
+
+			world.setBlockState(pos, state.withProperty(LIT, true));
+			if (stack.getItem().isDamageable()) {
+				stack.damageItem(1, player);
+			}
+			world.playSound(null, pos, SoundEvents.BLOCK_FIRE_AMBIENT, SoundCategory.PLAYERS, 1, 1);
+
+			return true;
+		} else if (state.getValue(LIT) && stack.getItem() == Items.WATER_BUCKET) {
+			if (world.isRemote) {
+				return true;
+			}
+
+			world.setBlockState(pos, state.withProperty(LIT, false));
+			if (!player.capabilities.isCreativeMode) {
+				player.setHeldItem(hand, new ItemStack(Items.BUCKET));
+			}
+			world.playSound(null, pos, SoundEvents.BLOCK_FIRE_EXTINGUISH, SoundCategory.PLAYERS, 0.5f, 1);
+		}
+
+		return super.onBlockActivated(world, pos, state, player, hand, facing, hitX, hitY, hitZ);
+	}
+
+	private boolean isFireStarter(ItemStack heldItem) {
+		return heldItem.getItem() == Items.FLINT_AND_STEEL || heldItem.getItem() == Item.getItemFromBlock(Blocks.TORCH);
+	}
+
+	@Override
+	public void getDrops(NonNullList<ItemStack> drops, IBlockAccess world, BlockPos pos, IBlockState state, int fortune) {
+		ItemStack BlockBrazierEmber = new ItemStack(this);
+		BlockBrazierEmber.setTagCompound(new NBTBuilder().setBoolean(LIT_TAG, false).build());
+		drops.add(BlockBrazierEmber);
+	}
+
+	@Override
+	public int tickRate(World worldIn) {
+		return 30;
+	}
+	@Override
```

Comment:
there should be new line here just to separate the previous block from override annotation